[BUG] prune_partition_column_test fails for Avro

Many tests in prune_partition_column_test.py are failing on Avro in the nightly builds.  Tests that are failing:


Test Name  ↑ | Duration | Age
-- | -- | --
Integration Tests 31x + 32x / spark-3.2.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[c-avro-True] | 0.84 sec | 1
Integration Tests 31x + 32x / spark-3.2.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[b-avro-True] | 0.78 sec | 1
Integration Tests 31x + 32x / spark-3.2.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[a-avro-True] | 1.5 sec | 1
Integration Tests 31x + 32x / spark-3.2.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project[avro-True] | 0.88 sec | 1
Integration Tests 31x + 32x / spark-3.1.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[c-avro-True] | 0.86 sec | 1
Integration Tests 31x + 32x / spark-3.1.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[b-avro-True] | 0.82 sec | 1
Integration Tests 31x + 32x / spark-3.1.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project_filter[a-avro-True] | 0.93 sec | 1
Integration Tests 31x + 32x / spark-3.1.3 on ubuntu20.04 w/ cuda11.2 / src.main.python.prune_partition_column_test.test_prune_partition_column_when_project[avro-True] | 0.9 sec | 1


Example failure stacktrace:
```
Caused by: java.lang.IllegalArgumentException: '-100: class java.lang.Long' is not supported for IntegerType, expecting Int.
	at com.nvidia.spark.rapids.GpuScalar$.from(literals.scala:295)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$5(GpuMultiFileReader.scala:254)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$5$adapted(GpuMultiFileReader.scala:252)
	at scala.collection.IndexedSeqOptimized.foreach(IndexedSeqOptimized.scala:36)
	at scala.collection.IndexedSeqOptimized.foreach$(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:198)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$4(GpuMultiFileReader.scala:252)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$4$adapted(GpuMultiFileReader.scala:251)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:55)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:53)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.withResource(GpuMultiFileReader.scala:158)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$3(GpuMultiFileReader.scala:251)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$3$adapted(GpuMultiFileReader.scala:249)
	at scala.collection.TraversableLike$WithFilter.$anonfun$foreach$1(TraversableLike.scala:877)
	at scala.collection.immutable.List.foreach(List.scala:392)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:876)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$buildPartitionsColumns$1(GpuMultiFileReader.scala:249)
	at com.nvidia.spark.rapids.Arm.closeOnExcept(Arm.scala:109)
	at com.nvidia.spark.rapids.Arm.closeOnExcept$(Arm.scala:107)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.closeOnExcept(GpuMultiFileReader.scala:158)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.buildPartitionsColumns(GpuMultiFileReader.scala:248)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.$anonfun$concatAndAddPartitionColsToBatchAndClose$1(GpuMultiFileReader.scala:237)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.withResource(GpuMultiFileReader.scala:158)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.concatAndAddPartitionColsToBatchAndClose(GpuMultiFileReader.scala:236)
	at com.nvidia.spark.rapids.MultiFileReaderUtils$.addMultiplePartitionValuesAndClose(GpuMultiFileReader.scala:208)
	at com.nvidia.spark.rapids.GpuColumnarBatchWithPartitionValuesIterator.next(GpuColumnarBatchIterator.scala:116)
	at com.nvidia.spark.rapids.GpuColumnarBatchWithPartitionValuesIterator.next(GpuColumnarBatchIterator.scala:95)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at com.nvidia.spark.rapids.FilePartitionReaderBase.get(GpuMultiFileReader.scala:376)
	at com.nvidia.spark.rapids.FilePartitionReaderBase.get(GpuMultiFileReader.scala:365)
	at com.nvidia.spark.rapids.PartitionIterator.next(dataSourceUtil.scala:39)
	at com.nvidia.spark.rapids.MetricsBatchIterator.next(dataSourceUtil.scala:49)
	at com.nvidia.spark.rapids.MetricsBatchIterator.next(dataSourceUtil.scala:43)
	at org.apache.spark.InterruptibleIterator.next(InterruptibleIterator.scala:40)
	at org.apache.spark.sql.rapids.GpuFileSourceScanExec$$anon$1.next(GpuFileSourceScanExec.scala:474)
	at org.apache.spark.sql.rapids.GpuFileSourceScanExec$$anon$1.next(GpuFileSourceScanExec.scala:463)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at com.nvidia.spark.rapids.ColumnarToRowIterator.$anonfun$fetchNextBatch$2(GpuColumnarToRowExec.scala:241)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.ColumnarToRowIterator.withResource(GpuColumnarToRowExec.scala:187)
	at com.nvidia.spark.rapids.ColumnarToRowIterator.fetchNextBatch(GpuColumnarToRowExec.scala:238)
	at com.nvidia.spark.rapids.ColumnarToRowIterator.loadNextBatch(GpuColumnarToRowExec.scala:215)
	at com.nvidia.spark.rapids.ColumnarToRowIterator.hasNext(GpuColumnarToRowExec.scala:255)
	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:345)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:498)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:501)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
```

in UCX (rapids_it-UCX-egx06-standalone) CI, we also found failures prune_partition_column_test for json cases,
```
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project[json-False]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project[json-True]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[a-json-False]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[a-json-True]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[b-json-False]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[b-json-True]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[c-json-False]
05:07:19  FAILED ../../src/main/python/prune_partition_column_test.py::test_prune_partition_column_when_project_filter[c-json-True]
```

which failed,
```
05:07:19  E                   : org.apache.spark.SparkException: Job aborted due to stage failure: Task 45 in stage 63592.0 failed 1 times, most recent failure: Lost task 45.0 in stage 63592.0 (TID 1870121) (10.136.6.4 executor 0): java.io.IOException: Error when processing file [path: file:///tmp/pyspark_tests/spark-egx-06-master-54268-826079085/PARTED_DATA/b=1/c=0/part-00006-372c52bc-74df-4030-bf12-e210a458d738.c000.json, range: 0-3, partition values: [0]]
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.$anonfun$readToTable$1(GpuJsonScan.scala:277)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.readToTable(GpuJsonScan.scala:272)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.readToTable(GpuJsonScan.scala:235)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.$anonfun$readToTable$4(GpuTextBasedPartitionReader.scala:328)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.readToTable(GpuTextBasedPartitionReader.scala:327)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.$anonfun$readBatch$1(GpuTextBasedPartitionReader.scala:275)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.readBatch(GpuTextBasedPartitionReader.scala:273)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.next(GpuTextBasedPartitionReader.scala:599)
05:07:19  E                   	at com.nvidia.spark.rapids.PartitionReaderWithBytesRead.next(dataSourceUtil.scala:62)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarPartitionReaderWithPartitionValues.next(ColumnarPartitionReaderWithPartitionValues.scala:36)
05:07:19  E                   	at com.nvidia.spark.rapids.PartitionReaderIterator.hasNext(PartitionReaderIterator.scala:37)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:93)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:173)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:93)
05:07:19  E                   	at org.apache.spark.sql.rapids.GpuFileSourceScanExec$$anon$1.hasNext(GpuFileSourceScanExec.scala:468)
05:07:19  E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.$anonfun$fetchNextBatch$2(GpuColumnarToRowExec.scala:239)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.withResource(GpuColumnarToRowExec.scala:187)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.fetchNextBatch(GpuColumnarToRowExec.scala:238)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.loadNextBatch(GpuColumnarToRowExec.scala:215)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.hasNext(GpuColumnarToRowExec.scala:255)
05:07:19  E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
05:07:19  E                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:345)
05:07:19  E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
05:07:19  E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
05:07:19  E                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
05:07:19  E                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
05:07:19  E                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
05:07:19  E                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
05:07:19  E                   	at org.apache.spark.scheduler.Task.run(Task.scala:131)
05:07:19  E                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
05:07:19  E                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
05:07:19  E                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
05:07:19  E                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
05:07:19  E                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
05:07:19  E                   	at java.lang.Thread.run(Thread.java:750)
05:07:19  E                   Caused by: ai.rapids.cudf.CudfException: CUDF failure at: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-317-cuda11/thirdparty/cudf/cpp/src/io/json/reader_impl.cu:648: Error determining column names.
05:07:19  E                   
05:07:19  E                   	at ai.rapids.cudf.Table.readJSON(Native Method)
05:07:19  E                   	at ai.rapids.cudf.Table.readJSON(Table.java:966)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.$anonfun$readToTable$1(GpuJsonScan.scala:274)
05:07:19  E                   	... 46 more
05:07:19  E                   
05:07:19  E                   Driver stacktrace:
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2258)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2207)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2206)
05:07:19  E                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
05:07:19  E                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
05:07:19  E                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2206)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1079)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1079)
05:07:19  E                   	at scala.Option.foreach(Option.scala:407)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1079)
05:07:19  E                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2445)
05:07:19  E                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2387)
05:07:19  E                   	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2376)
05:07:19  E                   	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
05:07:19  E                   	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:868)
05:07:19  E                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2196)
05:07:19  E                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2217)
05:07:19  E                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2236)
05:07:19  E                   	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2261)
05:07:19  E                   	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1030)
05:07:19  E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
05:07:19  E                   	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
05:07:19  E                   	at org.apache.spark.rdd.RDD.withScope(RDD.scala:414)
05:07:19  E                   	at org.apache.spark.rdd.RDD.collect(RDD.scala:1029)
05:07:19  E                   	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:390)
05:07:19  E                   	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:3519)
05:07:19  E                   	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:3687)
05:07:19  E                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$5(SQLExecution.scala:103)
05:07:19  E                   	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:163)
05:07:19  E                   	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:90)
05:07:19  E                   	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:775)
05:07:19  E                   	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:64)
05:07:19  E                   	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3685)
05:07:19  E                   	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:3516)
05:07:19  E                   	at sun.reflect.GeneratedMethodAccessor106.invoke(Unknown Source)
05:07:19  E                   	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
05:07:19  E                   	at java.lang.reflect.Method.invoke(Method.java:498)
05:07:19  E                   	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
05:07:19  E                   	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
05:07:19  E                   	at py4j.Gateway.invoke(Gateway.java:282)
05:07:19  E                   	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
05:07:19  E                   	at py4j.commands.CallCommand.execute(CallCommand.java:79)
05:07:19  E                   	at py4j.GatewayConnection.run(GatewayConnection.java:238)
05:07:19  E                   	at java.lang.Thread.run(Thread.java:750)
05:07:19  E                   Caused by: java.io.IOException: Error when processing file [path: file:///tmp/pyspark_tests/spark-egx-06-master-54268-826079085/PARTED_DATA/b=1/c=0/part-00006-372c52bc-74df-4030-bf12-e210a458d738.c000.json, range: 0-3, partition values: [0]]
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.$anonfun$readToTable$1(GpuJsonScan.scala:277)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.readToTable(GpuJsonScan.scala:272)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.readToTable(GpuJsonScan.scala:235)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.$anonfun$readToTable$4(GpuTextBasedPartitionReader.scala:328)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.readToTable(GpuTextBasedPartitionReader.scala:327)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.$anonfun$readBatch$1(GpuTextBasedPartitionReader.scala:275)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.withResource(GpuTextBasedPartitionReader.scala:203)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.readBatch(GpuTextBasedPartitionReader.scala:273)
05:07:19  E                   	at com.nvidia.spark.rapids.GpuTextBasedPartitionReader.next(GpuTextBasedPartitionReader.scala:599)
05:07:19  E                   	at com.nvidia.spark.rapids.PartitionReaderWithBytesRead.next(dataSourceUtil.scala:62)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarPartitionReaderWithPartitionValues.next(ColumnarPartitionReaderWithPartitionValues.scala:36)
05:07:19  E                   	at com.nvidia.spark.rapids.PartitionReaderIterator.hasNext(PartitionReaderIterator.scala:37)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:93)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.nextIterator(FileScanRDD.scala:173)
05:07:19  E                   	at org.apache.spark.sql.execution.datasources.FileScanRDD$$anon$1.hasNext(FileScanRDD.scala:93)
05:07:19  E                   	at org.apache.spark.sql.rapids.GpuFileSourceScanExec$$anon$1.hasNext(GpuFileSourceScanExec.scala:468)
05:07:19  E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.$anonfun$fetchNextBatch$2(GpuColumnarToRowExec.scala:239)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
05:07:19  E                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.withResource(GpuColumnarToRowExec.scala:187)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.fetchNextBatch(GpuColumnarToRowExec.scala:238)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.loadNextBatch(GpuColumnarToRowExec.scala:215)
05:07:19  E                   	at com.nvidia.spark.rapids.ColumnarToRowIterator.hasNext(GpuColumnarToRowExec.scala:255)
05:07:19  E                   	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:458)
05:07:19  E                   	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:345)
05:07:19  E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:898)
05:07:19  E                   	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:898)
05:07:19  E                   	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
05:07:19  E                   	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:373)
05:07:19  E                   	at org.apache.spark.rdd.RDD.iterator(RDD.scala:337)
05:07:19  E                   	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
05:07:19  E                   	at org.apache.spark.scheduler.Task.run(Task.scala:131)
05:07:19  E                   	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
05:07:19  E                   	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
05:07:19  E                   	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
05:07:19  E                   	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
05:07:19  E                   	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
05:07:19  E                   	... 1 more
05:07:19  E                   Caused by: ai.rapids.cudf.CudfException: CUDF failure at: /home/jenkins/agent/workspace/jenkins-spark-rapids-jni_nightly-dev-317-cuda11/thirdparty/cudf/cpp/src/io/json/reader_impl.cu:648: Error determining column names.
05:07:19  E                   
05:07:19  E                   	at ai.rapids.cudf.Table.readJSON(Native Method)
05:07:19  E                   	at ai.rapids.cudf.Table.readJSON(Table.java:966)
05:07:19  E                   	at org.apache.spark.sql.catalyst.json.rapids.JsonPartitionReader.$anonfun$readToTable$1(GpuJsonScan.scala:274)
05:07:19  E                   	... 46 more
```

@firestarman can you help check this failures? thanks

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AvroProviderImpl.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/AvroProviderImpl.scala
@@ -81,7 +81,7 @@ class AvroProviderImpl extends AvroProvider {
           broadcastedConf,
           fileScan.relation.dataSchema,
           fileScan.requiredSchema,
-          fileScan.relation.partitionSchema,
+          fileScan.readPartitionSchema,
           new AvroOptions(fileScan.relation.options, broadcastedConf.value.value),
           fileScan.allMetrics,
           pushedFilters,

--- a/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
+++ b/sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuFileSourceScanExec.scala
@@ -89,7 +89,8 @@ case class GpuFileSourceScanExec(
     dataOutAttrs ++ prunedPartOutAttrs
   }.getOrElse(originalOutput)
 
-  private val readPartitionSchema = requiredPartitionSchema.getOrElse(relation.partitionSchema)
+  private[rapids] val readPartitionSchema =
+    requiredPartitionSchema.getOrElse(relation.partitionSchema)
 
   // this is set only when we either explicitly replaced a path for CONVERT_TIME
   // or when TASK_TIME if one of the paths will be replaced.

